### ISSUE TITLE
Missing identity for logaddexp

### ISSUE THREAD

**Issue by chebee7i** [numpy/numpy#4599]
Should `logaddexp` and `logaddexp2` have identities defined for reduce?

```
In [1]: x = np.array([-1,-1,-1], dtype=float)

In [2]: np.logaddexp2.reduce(x)
Out[2]: 0.58496250072115619

In [3]: np.logaddexp2.reduce(x[:2])  # matches np.logaddexp2(*x[:2])
Out[3]: 0.0

In [4]: np.logaddexp2.reduce(x[:1])
Out[4]: -1.0

In [5]: np.logaddexp2.reduce(x[:0])
ValueError: zero-size array to reduction operation logaddexp2 which has no identity
```

I was expecting `-inf` when reducing an empty array.

**Comment by souravsingh** [comment]
I would like to work on the issue. How do I start?

**Comment by Ellest** [comment]
Is this still open for someone to pick up? I'm new to this and wasn't sure if "Open" status means open for anyone... Thanks.

**Comment by eric-wieser** [comment]
Usually yes, but in this case there's already a patch awaiting review at #8955. That doesn't mean you can't submit a different patch, if you think you can do it better...

**Comment by arogozhnikov** [comment]
Just faces this.
```
In [6]: np.logaddexp.reduce(np.arange(5))
TypeError: No loop matching the specified signature was found for ufunc logaddexp
```

Very confusing behavior. Is there any progress with this issue?

**Comment by mattip** [comment]
Fixed by #8955
